### ISSUE TITLE
How do I extract layers and their "functionality" from pre-trained network?

### ISSUE THREAD

**Issue by tz-hmc** [apache/incubator-mxnet#8594]
## Description: 
I need to make a Siamese network for image similarity, and to do so I'm hoping to use 2 copies of Resnet152. Sorry, I made a separate post earlier about a slightly similar problem (about sharing weights). However, after reading a few more related issues, it seems like the best option is to make new layers and copy the weights over, set the weights to the same variable (for weight sharing) and then fine-tune it. 

So now my question is: How do I do that? I can get weights from arg_params but how do I know whether the layer is convolutional/fully connected/etc? Also, can the Load initializer load into a variable or does it have to load into a model? 

I also can't find how to get the names from Symbol and am pretty hopeless with reading documentation tbh, and couldn't figure out where this could be supported. 

Here's how I'm imagining it to work, but I'm sure it doesn't (error says the convolution layer doesn't have the right number of inputs or something):

```
sym, arg_params, aux_params = mx.model.load_checkpoint('resnet-152', 0)
layer = sym.get_internals()
# Load pretrained model
# How do I ensure that the weights are the same? Will this way work?
sym, arg_params, aux_params = mx.model.load_checkpoint('resnet-152', 0)
layer = sym.get_internals()
print type(sym)
print type(arg_params), len(arg_params)
print type(aux_params), len(aux_params)

# initialize
nn_layer_1 = [mx.sym.Variable('input_1')]
nn_layer_2 = [mx.sym.Variable('input_2')]
for name, args in arg_params.items():
    var = mx.sym.Variable(name+'_1')
    var = mx.initializer.Load(args.asnumpy()) # ??? how to initialize specific variable
    in1 = nn_layer_1[-1]
    in2 = nn_layer_2[-1]
    # initialize var to args
    nn_layer_1.append(mx.sym.Convolution(in1, weight=var, no_bias=True))
    nn_layer_2.append(mx.sym.Convolution(in2, weight=var, no_bias=True))

# fe_sym = mx.sym.Concat(layer1['flatten0_output'], layer2['flatten0_output_2'],dim=0)
fe_sym = mx.sym.Concat(nn_layer_1[-1], nn_layer_2[-1],dim=0)
```

Please help, and thanks in advance!

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by indhub** [comment]
This is not an issue. For how-to questions, you are much more likely to get quick answers in the [user forum](https://discuss.mxnet.io/). Thanks!
